dartagnan/src/main/java/com/dat3m/dartagnan/wmm/axiom/Acyclic.java

```diff
@@ -33,16 +49,159 @@ public Acyclic(Relation rel) {
         super(rel, false, false);
     }
 
+    public static final class Statistics {
+        private final List<Axiom> axioms;
+        private final ExecutionAnalysis exec;
+        private final RelationAnalysis ra;
+
+        public Statistics(List<Axiom> a, Context c) {
+            axioms = a;
+            exec = c.get(ExecutionAnalysis.class);
+            ra = c.get(RelationAnalysis.class);
+        }
+
+        @Override
+        public String toString() {
+            int i = 0;
+            for (Axiom a : axioms) {
+                if (a instanceof Acyclic) {
+                    i += ((Acyclic) a).getEncodeTupleSet(exec, ra).size();
+                }
+            }
+            return "Number of encoded tuples for acyclicity: " + i;
+        }
+    }
```

Comment:
I find this class awkward. Is the only reason it is placed here to access `getEncodeTupleSet`?
If that is the case, then a simple `getEncodingSize` function would be easier and more flexibie, because the caller can decide which axioms to track and how to present/print it.